Nonprofit-Exchange-Hub/web-app
# 343 add validation to user signup form

## Dev Summary

[FIgma User SIgn Up flow](https://www.figma.com/file/K7DT5jYcLKRc9epmNrO0N7/UX---Dev-Handoff?type=design&node-id=200-5472&mode=design&t=PrW6vZ6JdjCKcoDN-0)


This adds validation to the fields in the User Sign Up form.  It also refactors the UI into multiple step components and updates the display of errors to be under the field instead of under the label.  This also adds `redux` for state management for `user` entity, and `redux-thunk` for sending the `POST` request to the backend to create the user account.

The work was starting to have scope-creep, so created some follow-up tickets:
- https://github.com/orgs/Nonprofit-Exchange-Hub/projects/14/views/1?pane=issue&itemId=34900098
- https://github.com/orgs/Nonprofit-Exchange-Hub/projects/14/views/1?pane=issue&itemId=34900640
- https://github.com/orgs/Nonprofit-Exchange-Hub/projects/14/views/1?pane=issue&itemId=34901276


## Test Plan

To test, go to the User Signup form via `http://localhost:3000/signup-citizen`.
- try to go through the form by entering invalid or missing values and see if proper error messages are displayed
- verify that when all fields are filled with valid information, the user reaches the email verification step

Note:
During account creation on my local, I was getting an error regarding `SendGrid` - https://github.com/Nonprofit-Exchange-Hub/web-app/blob/cc54d3661848274cd6cae0df160f6917a463cbbc/server/src/acccount-manager/account-manager.controller.ts#L105 ... I don't think I have a valid `SENDGRID_API_KEY` in my `.env`.

React-query tutorial: https://www.youtube.com/watch?v=r8Dg0KVnfMA
